src/libraries/System.Private.CoreLib/src/System/Single.cs

```diff
@@ -435,6 +435,33 @@ public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatPro
             return Number.TryParseFloat(s, style, NumberFormatInfo.GetInstance(provider), out result);
         }
 
+        //
+        // Helpers, those methods are referenced from the JIT
+        //
+
+        private static float Remainder(float dividend, float divisor)
+        {
+            // From the ECMA standard:
+            //
+            // If [divisor] is zero or [dividend] is infinity
+            //   the result is NaN.
+            // If [divisor] is infinity,
+            //   the result is [dividend] (negated for -infinity***).
+            //
+            // ***"negated for -infinity" has been removed from the spec
+            if (divisor == 0 || !IsFinite(dividend))
+            {
+                return NaN;
+            }
+
+            if (!IsFinite(divisor) && !IsNaN(divisor))
+            {
+                return dividend;
+            }
```

Comment:
Is this needed? C requires (for IEC 60559; aka IEEE 754; compatible implementations, which all our dependencies have to be):
* If x is ±0 and y is not zero, ±0 is returned.
* If x is ±∞ and y is not NaN, NaN is returned and FE_INVALID is raised.
* If y is ±0 and x is not NaN, NaN is returned and FE_INVALID is raised.
* If y is ±∞ and x is finite, x is returned.
* If either argument is NaN, NaN is returned.

So it looks like we can just defer to `FMod` directly and bypass this extra logic

Reply:
`If y is ±∞ and x is finite, x is returned.`
The `x is finite` part doesn't seem to be checked here so it'd different it seems.

Reply:
Values can be `finite`, `infinite`, or `nan`.

The implementation here is doing:
* If `divisor` (`y`) is zero, return nan
  * i.e. the third/fifth bullet points
* if `dividend` (`x`) is not finite (in other words, `x` is `nan` or `infinite`), return `NaN`
  * i.e. the second/fifth bullet points
* if `divisor` (`y`) is not `finite` and not `nan` (in other words, `y` is `infinite`), return `dividend` (`x`)
  * i.e, the 4th bullet point
  * `x` being `nan` or `infinite` is covered by the check I just described previously

Everything else defers down to `fmod`

Reply:
Ah I missed that it's `||` not `&&` there, gonna make it use FMod directly then.

Reply:
I've made the helper be `Math.FMod` directly in the last commit, in case it'll be rewritten in C# it should be then added back in `jitinterface.cpp` to have the JIT emit direct calls to it.